### PR TITLE
Feature/add onnxrt acl execution provider

### PR DESCRIPTION
**Pull request by jafet-chaves**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -59,13 +59,28 @@ RR_CHECK_FEATURE_LIB(TFLITE, TensorFlow lite Installation,
...
                      onnxruntime, OrtGetApiBase, onnxruntime/core/session/onnxruntime_cxx_api.h, no)
+
+AC_ARG_ENABLE(onnxrt_acl,[
+AS_HELP_STRING([--enable-onnxrt-acl],
+[Build ONNXRT backend with Arm Computer Library (ACL) execution provider (disabled by default)])
+])
+
+if test x$enable_onnxrt_acl = xyes; then
+    AC_MSG_NOTICE([Building the ONNXRT backend with ACL support])
+    AM_COND_IF([HAVE_ONNXRT], [], [AC_MSG_ERROR(The ONNXRT ACL needs ONNRT backend enabled aswell)])
```
Comment (jafet-chaves) [inline configure.ac:78]: Fix typo

File: configure.ac [hunk 2]
```diff
@@ -59,13 +59,28 @@ RR_CHECK_FEATURE_LIB(TFLITE, TensorFlow lite Installation,
...
+
+    AC_SUBST([ONNXRT_LIBS], ["$ONNXRT_LIBS -larm_compute -larm_compute_core -larm_compute_graph"])
+    AC_DEFINE(HAVE_ONNXRT_ACL, 1, [Define this to build the ACL support])
```
Comment (jafet-chaves) [inline configure.ac:81]: Change to "Define this to build the ACL support for ONNXRT backend"

File: r2i/onnxrt/engine.cc [hunk 3]
```diff
@@ -122,6 +129,7 @@ RuntimeError Engine::Start ()  {
   try {
     this->CreateEnv();
     this->CreateSessionOptions();
+    this->AppendSessionOptionsExecutionProvider(this->session_options, error);
```
Comment (lumurillo) [inline r2i/onnxrt/engine.cc:132]: Could this be called from the AppendSessionOptionsExecutionProvider function?

File: configure.ac [hunk 4]
```diff
@@ -59,13 +59,21 @@ RR_CHECK_FEATURE_LIB(TFLITE, TensorFlow lite Installation,
...
                      nvinfer, createInferBuilder_INTERNAL, NvInfer.h, no)
 
 RR_CHECK_FEATURE_LIB(ONNXRT, ONNX Runtime Installation,
                      onnxruntime, OrtGetApiBase, onnxruntime/core/session/onnxruntime_cxx_api.h, no)
+
+RR_CHECK_FEATURE_LIB(ONNXRT_ACL, ONNX Runtime Installation with Arm Compute Library support,
+                     onnxruntime, OrtSessionOptionsAppendExecutionProvider_ACL, onnxruntime/core/providers/acl/acl_provider_factory.h, no)
+AM_COND_IF([HAVE_ONNXRT_ACL], [AM_COND_IF([HAVE_ONNXRT], [],
+ [AC_MSG_ERROR(The ONNXRT ACL needs ONNXRT backend enabled aswell)])], [])
+AC_SUBST([ONNXRT_ACL_LIBS], ["$ONNXRT_LIBS -larm_compute -larm_compute_core -larm_compute_graph"])
```
Comment (michaelgruner) [inline configure.ac:75]: We need to check these libraries exist before attempting to link them.

File: meson.build [hunk 5]
```diff
@@ -96,6 +96,23 @@ if get_option('enable-onnxrt')
   cdata.set('HAVE_ONNXRT', 1)
 endif
 
+# Define library dependencies for ONNXRT with ACL execution provider support
+if get_option('enable-onnxrt-acl')
+  if onnxrt_dep.found()
+    arm_compute = cpp.find_library('arm_compute', required: true)
+    arm_compute_dep = declare_dependency(dependencies: arm_compute)
+    arm_compute_core = cpp.find_library('arm_compute', required: true)
```
Comment (michaelgruner) [inline meson.build:104]: you h ave the name of the library wrong.

File: meson.build [hunk 6]
```diff
@@ -96,6 +96,23 @@ if get_option('enable-onnxrt')
   cdata.set('HAVE_ONNXRT', 1)
 endif
 
+# Define library dependencies for ONNXRT with ACL execution provider support
+if get_option('enable-onnxrt-acl')
+  if onnxrt_dep.found()
```
Comment (michaelgruner) [inline meson.build:101]: This dep is not defined if ONNXRT was not selected.

File: r2i/onnxrt_acl/engine.cc [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+void Engine::AppendSessionOptionsExecutionProvider(Ort::SessionOptions
+    &session_options, r2i::RuntimeError &error) {
+  OrtStatus *status = NULL;
+
+  status = OrtSessionOptionsAppendExecutionProvider_ACL(session_options,
+           USE_ARENA);
+  if (status != NULL) {
+    error.Set (RuntimeError::Code::FRAMEWORK_ERROR,
+               "Failed setting Arm Computer Library (ACL) execution provider");
```
Comment (michaelgruner) [inline r2i/onnxrt_acl/engine.cc:33]: There is a message in the status, I recommend using that.

File: r2i/onnxrt_acl/engine.cc [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+Engine::Engine () : onnxrt::Engine() {
+
+}
+
+void Engine::AppendSessionOptionsExecutionProvider(Ort::SessionOptions
+    &session_options, r2i::RuntimeError &error) {
+  OrtStatus *status = NULL;
+
+  status = OrtSessionOptionsAppendExecutionProvider_ACL(session_options,
+           USE_ARENA);
```
Comment (michaelgruner) [inline r2i/onnxrt_acl/engine.cc:30]: Eventually we want to expose the arena as a parameter.

File: r2i/onnxrt_acl/engine.h [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#ifndef R2I_ONNXRT_ACL_ENGINE_H
+#define R2I_ONNXRT_ACL_ENGINE_H
+
+#include <r2i/onnxrt/engine.h>
+
+#include <core/providers/acl/acl_provider_factory.h>
```
Comment (michaelgruner) [inline r2i/onnxrt_acl/engine.h:17]: This is not used in the header (which is a good thing). Move this to the source file.

File: r2i/onnxrt/engine.cc [hunk 10]
```diff
@@ -9,11 +9,16 @@
  * back to RidgeRun without any encumbrance.
  */
 
+#include "config.h"
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/engine.cc]: Outdated, this code doesn't represent the latest changes.

File: r2i/onnxrt/engine.cc [hunk 11]
```diff
@@ -9,11 +9,16 @@
...
 #include "r2i/onnxrt/engine.h"
 
 #include <core/common/exceptions.h>
 #include <core/session/onnxruntime_cxx_api.h>
 
+#ifdef HAVE_ONNXRT_ACL
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/engine.cc]: Outdated, this code doesn't represent the latest changes.

**Pull request merged**